os/arch/arm/src/armv7-a/arm_cpustart.c

```diff
@@ -163,27 +168,54 @@ int arm_start_handler(int irq, void *context, void *arg)
  * Returned Value:
  *   Zero on success; a negated errno value on failure.
  *
+ * Assumptions:
+ *   - Called from a different CPU than the target
+ *   - Target CPU is currently powered off and can be safely started
+ *   - System is in a state to safely bring additional CPU online
+ *
  ****************************************************************************/
 
 int up_cpu_start(int cpu)
 {
-	svdbg("Starting CPU%d\n", cpu);
+    int ret = OK;
+
+    svdbg("Starting CPU%d\n", cpu);
 
 	DEBUGASSERT(cpu >= 0 && cpu < CONFIG_SMP_NCPUS && cpu != this_cpu());
 
+    /* take the lock, as we are about to put cpu idle task back active */
+	sched_lock();
```

Comment:
We always fix the idle thread to be located at the end of the cpu assignment queue.

Therefore, we do not have to hold the sched_lock  for idle thread in the cpu on/off process.

If you are concerned that the scheduling of the cpu occurs during the cpu on/off process,

 it will be added because it should be protected by sched_lock and enter_critical_section from the point before using this api, so you do not have to protect it with sched_lock here.